src/qililab/platform/platform.py

```diff
@@ -1590,3 +1592,119 @@ def draw(self, qprogram: QProgram, time_window: int | None = None, averages_disp
         result = qblox_draw.draw(sequencer, runcard_data, time_window, averages_displayed)
 
         return result
+
+    def stream_array(
+        self,
+        shape: tuple,
+        loops: dict[str, np.ndarray],
+        experiment_name: str,
+        db_manager: DatabaseManager,  # get rid
+        qprogram: QProgram | None = None,
+        optional_identifier: str | None = None,
+    ):
+        """Allows for real time saving of results from an experiment.
+
+        This class wraps a numpy array and adds a context manager to save results and database metadata on real time
+        while they are acquired by the instruments.
+
+        Example usage of this function:
+
+            .. code-block:: python
+
+                db_manager = ql.get_db_manager()
+                db_manager.set_sample_and_cooldown(sample=sample, cooldown=cooldown)
+
+                platform = ql.build_platform(runcard=runcard)
+                platform.connect()
+                platform.initial_setup()
+                platform.turn_on_instruments()
+
+                (experiment definition)
+
+                stream_array = platform.stream_array(
+                    shape=(len(if_sweep), 2),
+                    loops={"frequency": if_sweep},
+                    platform=platform,
+                    experiment_name="resonator_spectroscopy_cw",
```

Comment:
```suggestion
                    experiment_name="resonator_spectroscopy",
```

Sounds like a picky request I know but to start going back to documenting the library for external users let's keep the most simple name of the experiment.